### PR TITLE
add new holdingpen component

### PR DESCRIPTION
**Pull request by jmartinm**
* Adds new holdingpen editor component that deals with opening records
  from the Holding Pen.

Signed-off-by: Javier Martin Montull <javier.martin.montull@cern.ch>

### CODE REVIEW

File: src/app/shared/services/holdingpen.service.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import { AppConfig } from '../../app.config';
+
+@Injectable()
+export class HoldingPenService {
+  private objectid: string;
```
Comment (harunurhan) [inline src/app/shared/services/holdingpen.service.ts]: It may be better, if it is `camelCase` (`objectId`) like in the config.

**Comment by jmartinm** [comment]
I have done the changes and removed the HoldingPen service in the end. Can you have another look?

File: src/app/editor-container/editor-container.component.ts [hunk 2]
```diff
@@ -48,9 +47,9 @@ export class EditorContainerComponent implements OnInit {
       .subscribe(params => {
         this.recordService.fetchRecord(params['type'], params['recid'])
           .then(record => {
-            this.record = record;
+            this.record = record['metadata'];
             this.config = this.appConfig.getConfigForRecord(record);
-            return this.recordService.fetchSchema(record['$schema']);
+            return this.recordService.fetchUrl(record['$schema']);
```
Comment (harunurhan) [inline src/app/editor-container/editor-container.component.ts]: shouldn't this be `this.record['$schema']` or `record['metadata']['$schema']`

File: src/app/shared/services/record.service.ts [hunk 3]
```diff
@@ -31,29 +31,35 @@ import { AppConfig } from '../../app.config';
...
+  }
+
   saveRecord(record: Object): Observable<Object> {
     return this.http
       .put(this.config.apiUrl(this.pidType, this.pidValue), record)
       .map(res => res.json());
   }
+
+  saveHoldingPenObject(record: Object): Observable<Object> {
+    return this.http.put(this.config.holdingPenApiUrl(this.objectId), record).map(res => res.json());
```
Comment (harunurhan) [inline src/app/shared/services/record.service.ts]: better if you put operation per line like `saveRecord`

File: src/app/shared/services/record.service.ts [hunk 4]
```diff
@@ -31,29 +31,35 @@ import { AppConfig } from '../../app.config';
 export class RecordService {
```
Comment (harunurhan) [inline src/app/shared/services/record.service.ts]: Now that this service does everything maybe better to rename it to `ApiCallService` or `ApiService` or `ApiClientService` or something similar.

File: src/app/shared/services/record.service.ts [hunk 5]
```diff
@@ -31,29 +31,35 @@ import { AppConfig } from '../../app.config';
...
   private pidType: string;
   private pidValue: string;
-  private config: AppConfig;
+  private objectId: string;
```
Comment (harunurhan) [inline src/app/shared/services/record.service.ts]: An empty line above to separate from others. Also single comments to explain these properties would be nice.

File: src/app/editor-holdingpen/editor-holdingpen.component.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    private recordService: RecordService,
+    private appConfig: AppConfig) { }
+
+  ngOnInit() {
+    this.route.params
+      .subscribe(params => {
+        this.recordService.fetchHoldingPenObject(params['objectid'])
+          .then(record => {
+            this.workflowObject = record;
+            this.record = record['metadata'];
```
Comment (harunurhan) [inline src/app/editor-holdingpen/editor-holdingpen.component.ts]: I don't think we need to keep `this.record`  so in template, just use `[record] = "workflowObject.metadata"`

File: src/app/editor-holdingpen/editor-holdingpen.component.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+  constructor(private route: ActivatedRoute,
+    private recordService: RecordService,
+    private appConfig: AppConfig) { }
+
+  ngOnInit() {
+    this.route.params
+      .subscribe(params => {
+        this.recordService.fetchHoldingPenObject(params['objectid'])
+          .then(record => {
```
Comment (harunurhan) [inline src/app/editor-holdingpen/editor-holdingpen.component.ts]: better rename anonymous function param, `record` to name of whatever was fetched

File: src/app/editor-holdingpen/editor-holdingpen.component.ts [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+@Component({
+  templateUrl: './editor-holdingpen.component.html',
+  styleUrls: [
+    './editor-holdingpen.component.scss'
+  ]
+})
+export class EditorHoldingPenComponent implements OnInit {
+  private record: Object;
+  private workflowObject: Object;
```
Comment (harunurhan) [inline src/app/editor-holdingpen/editor-holdingpen.component.ts]: To make it uniform, I think we should use either `holdingPenObject` or `workflowObject` everywhere.

**Pull request merged**